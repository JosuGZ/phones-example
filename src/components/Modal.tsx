import * as React from 'react';

export interface ModalProps {
  title: JSX.Element | string;
  body: JSX.Element | string;
  onClose: () => void;
}

export class Modal extends React.Component<ModalProps, undefined> {

  shouldComponentUpdate(nextProps: ModalProps) {
    var updated = false;

    if (this.props.title !== nextProps.title) updated = true;
    if (this.props.body !== nextProps.body) updated = true;

    return updated;
  }

  render() {
    return (
      <div onClick={this.props.onClose} className="modal-container">
        <div className="modal-window">
          <div className="modal-window-header">
            <div className="title">{this.props.title}</div>
            <button onClick={this.props.onClose} className="close">X</button>
          </div>
          <div className="modal-window-body">
            {this.props.body}
          </div>
        </div>
      </div>
    );
  }

}
