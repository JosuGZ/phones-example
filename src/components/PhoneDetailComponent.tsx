import * as React from 'react';

import { Modal } from './Modal';

export interface PhoneDetailComponentProps {
  phone: any;
  onClose: () => void;
}

export class PhoneDetailComponent extends React.Component<PhoneDetailComponentProps, undefined> {

  shouldComponentUpdate(nextProps: PhoneDetailComponentProps) {
    var updated = false;

    if (this.props.phone !== nextProps.phone) updated = true;

    return updated;
  }

  render() {
    var body = <pre>{JSON.stringify(this.props.phone, null, 2)}</pre>;
    var title = this.props.phone.name;

    return (
      <Modal title={title} body={body} onClose={this.props.onClose} />
    );
  }

}
