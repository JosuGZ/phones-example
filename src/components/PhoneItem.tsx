import * as React from 'react';

import { PhoneDetailComponent } from './PhoneDetailComponent';

export interface PhoneItemProps {
  phone: any;
}

interface PhoneItemState {
  showDetails: boolean;
}
export class PhoneItem extends React.Component<PhoneItemProps, PhoneItemState> {

  constructor(props: PhoneItemProps) {
    super(props);

    this.state = {
      showDetails: false
    };

    this.openDetails = this.openDetails.bind(this);
    this.closeDetails = this.closeDetails.bind(this);
  }

  shouldComponentUpdate(nextProps: PhoneItemProps, nextState: PhoneItemState) {
    var updated = false;

    if (this.props.phone !== nextProps.phone) updated = true;
    if (this.state.showDetails !== nextState.showDetails) updated = true;

    return updated;
  }

  openDetails() {
    this.setState({
      showDetails: true
    })
  }

  closeDetails() {
    this.setState({
      showDetails: false
    })
  }

  render() {
    var props = this.props;
    var state = this.state;

    return (
      <div className="phone-outer-box">
        <div className="phone" onClick={this.openDetails}>
          <div className="img-container">
            <img src={props.phone.image} />
          </div>
          <div className="name">{props.phone.name}</div>
        </div>
        { state.showDetails && <PhoneDetailComponent phone={props.phone} onClose={this.closeDetails} />}
      </div>
    );
  }

}
