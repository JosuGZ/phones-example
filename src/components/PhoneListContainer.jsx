import * as React from 'react';
import { connect } from 'react-redux';

import { PhoneItem } from './PhoneItem';

// export interface PhoneListContainerProps {
//   phones: any[];
// }

export class PhoneListContainer extends React.Component {
  render() {
    return (
      <div className="phone-list-container">{
        this.props.phones.map((phone, index) => <PhoneItem key={index} phone={phone} />)
      }</div>
    );
  }
}
