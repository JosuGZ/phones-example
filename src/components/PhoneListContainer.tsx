import * as React from 'react';
import { connect } from 'react-redux';

import { PhoneItem } from './PhoneItem';

export interface PhoneListContainerProps {
  phones: any[];
}

export const PhoneListContainer = function(props: PhoneListContainerProps) {
  return (
    <div className="phone-list-container">{
      props.phones.map((phone, index) => <PhoneItem key={index} phone={phone} />)
    }</div>
  );
}
