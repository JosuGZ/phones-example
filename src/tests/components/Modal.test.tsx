import * as React from 'react';
import { shallow } from 'enzyme';
import { Modal } from '../../components/Modal';

describe('Component Modal', () => {
  const title = 'Title';
  const body = 'Body';
  const onClose = () => {};
  const modal = shallow(<Modal title={title} body={body} onClose={onClose}/>);

  it('match snapshot', () => {
    expect(modal).toMatchSnapshot();
  });

});
