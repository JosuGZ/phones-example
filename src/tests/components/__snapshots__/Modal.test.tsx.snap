// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Modal match snapshot 1`] = `
<div
  className="modal-container"
  onClick={[Function]}
>
  <div
    className="modal-window"
  >
    <div
      className="modal-window-header"
    >
      <div
        className="title"
      >
        Title
      </div>
      <button
        className="close"
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="modal-window-body"
    >
      Body
    </div>
  </div>
</div>
`;
